File: app/src/main/java/com/mycompany/myapp/monitoring/LoggerModule.java
package com.mycompany.myapp.monitoring;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import timber.log.Timber.DebugTree;
import timber.log.Timber.Tree;

@Module
public class LoggerModule {
    @Singleton
    @Provides
    public Tree provideTimberTree() {
        return new DebugTree();
    }
}

File: app/src/debug/java/com/mycompany/myapp/monitoring/LoggerModule.kt
package com.mycompany.myapp.monitoring

import dagger.Module
import dagger.Provides
import timber.log.Timber
import timber.log.Timber.DebugTree
import javax.inject.Singleton

@Module
class LoggerModule {
    @Singleton
    @Provides
    fun provideTimberTree() : Timber.Tree = DebugTree()
}
